flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/committables/CommittableManager.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.operators.sink.committables;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.connector.sink2.Committer;
+import org.apache.flink.streaming.api.connector.sink2.CommittableSummary;
+import org.apache.flink.streaming.api.connector.sink2.CommittableWithLineage;
+
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * Internal wrapper to handle the committing of committables.
+ *
+ * @param <CommT> type of the committable
+ */
+@Internal
+public interface CommittableManager<CommT> {
```

Comment:
Hi @fapaul~ Sorry I might not fully got here: It seems currently we are using the method that treats the endOfInput as the checkpoint with id = Long.MAX_VALUE in the implementation? It seems to me that it is sufficient to handle both the streaming and batch cases~? Do we expected to have another `CommittableManager` implementation in the future~?

Reply:
I agree the implementation handles both similarly but the interfaces separate both concerns. Since the operators are in a separate module I think it is a good idea to only expose the interactions through interfaces and do not expose the implementation. 

Having two interfaces allows us in the future to maybe refactor the manager implementation and have different implementations for batch and streaming.

Reply:
Thanks for the explanation! I still think it would be better to not introduce the interfaces if it is not used for now, but I do not strongly argue for that for now.